ollama/_client.py

```diff
@@ -113,6 +113,15 @@ class Client(BaseClient):
   def __init__(self, host: Optional[str] = None, **kwargs) -> None:
     super().__init__(httpx.Client, host, **kwargs)
 
+  def close(self):
```

Comment:
You should update `BaseClient` to subclass `contextlib.AbstractContextManager` and `contextlib.AbstractAsyncContextManager`. Since these abstract base classes already implement `__enter__` and `__aenter__` by returning `self`, all you need to do is implement `__exit__` and `__aexit__`

Reply:
Seems like `__aenter__` is not implemented either

Reply:
I still like the version with contextlib more as it just tucks these away in the base client

Reply:
You need to also subclass `contextlib.AbstractAsyncContextManager`